functions/src/events/events.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * Events
+    Workshop 14
+        Title
+        Description
+        Instructions: []
+        Host: uid
+        Active: true
+        Dates: {
+	        Start: timestamp
+	        End: timestamp
+        }
+        Tags: [],
+        Category: [],
+        Online: boolean
+        Stats: {
+            Likes: []
+            Rsvp: []
+            people_who_showed_up:
+        people_who_stayed_till_the_end:
+        }
+        Location: {
+            Geo: {
+                Geopoint: geopoint,
+                Geohash: geohash
+        },
+        Address: {
+            fullAddress: string,
+            fullStreet: string,
+            streetNum: string,
+            Street: string,
+            City: string,
+            State: string?,
+            Country: String
+        }
+    }
+
+ */
+
+import { firestore } from "../admin/admin";
+
+interface event {
+    title: string;
+    description: string;
+    instructions: [];
+    host: number;
+    active: boolean;
+    tags: [];
+    category: [];
+    online: boolean;
+    stats: {"likes" : [], "rsvp" : [], "showedUp" : number, "stayed" : number};
```

Comment:
Its typically recommended that one use a type to define an object within an object. 

Kinda like this. Before the event interface create a stats type
```
type stats = { 
 likes: string[],
 rsvp: string[],
 showedUp: number,
 stayed: number
}
```

And then inside the event interface you do this
```
 stats: stats;
```

The reason for this is because people may want to use the sub-object stats by itself in certain occasions and its good practice to define a type for them.

Additional resource and examples: https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases